MSBuild parser does not recognize warnings after JENKINS-56613

### Jenkins and plugins versions report

```
Jenkins: 2.528.2
OS: Linux - 6.1.0-30-amd64
Java: 21.0.9 - Eclipse Adoptium (OpenJDK 64-Bit Server VM)
analysis-model-api:13.18.0-935.v784ca_107400a_
...
```

Since November 2025 we have the problem, that the MSBuild parser does not work properly anymore. Since we updated from 3.11 to 3.15 this problem exists and seems to be related to #1330 . I have let ChatGPT analyze the problem and think it detected a problem in the implementation. Here is the bugreport:

### Summary

After upgrading `analysis-model-api` from `13.11.0-919.vda_c75b_c73b_cc` to `13.15.0-928.v63f6dc44cfd2`, MSBuild warnings are no longer detected by Warnings NG in our Jenkins jobs.

`warnings-ng` itself stayed unchanged:
- working:
  - warnings-ng: `12.9848.v5603977006f2`
  - analysis-model-api: `13.11.0-919.vda_c75b_c73b_cc`
- broken:
  - warnings-ng: `12.9848.v5603977006f2`
  - analysis-model-api: `13.15.0-928.v63f6dc44cfd2`

So this looks like a regression in the underlying MSBuild parser shipped via `analysis-model-api`.

### What we see

The Jenkins build log clearly contains valid MSBuild warning lines, but no warnings are reported anymore.

Example line from the persisted raw job log:

```
[2025-11-24T09:30:26.235Z]     34>d:\****\workspace\MPA_master\ramdisk\mpa-clients\target\sources\MPAClient\WPF\Services\WpfMessageBoxService.cs(26,57): warning CS8625: Ein NULL-Literal kann nicht in einen Non-Nullable-Verweistyp konvertiert werden. [d:\****\workspace\MPA_master\ramdisk\mpa-clients\target\sources\MPAClient\MPAClient_egz53oxb_wpftmp.csproj]^M
```

### Expected behavior

That line should be parsed as regular MSBuild warning, just like they were with `analysis-model-api 13.11.0-919.vda_c75b_c73b_cc`.

### Actual behavior

No warning is created at all.

### Suspected root cause

This looks like a regression in `MsBuildParser` between the versions above.

Our raw Jenkins logs contain file paths with `*`, e.g.:

`d:\****\workspace\...`

In the newer parser, paths containing `*` appear to be treated as invalid path names, normalized to `"-"`, and then discarded as tool names / invalid file names. As a result, otherwise valid MSBuild warning lines are dropped completely.

This would explain why:
- the exact same Jenkins job worked before with `analysis-model-api 13.11.0`
- it stopped working immediately after updating only `analysis-model-api`
- `warnings-ng` itself did not change

### Reproduction idea

A minimal reproduction should be possible with an MSBuild warning line like this:

```
[2025-11-24T09:30:26.235Z]     34>d:\****\workspace\foo\Bar.cs(26,57): warning CS8625: Example warning text [d:\****\workspace\foo\foo.csproj]^M
```

If the parser treats the `*` inside the path as invalid and rejects the issue, the warning disappears.

### Notes

The `****` are present in the persisted raw Jenkins job log, not only in browser output.

Could you please check whether the MSBuild parser should still accept such lines, or at least avoid discarding the whole issue when masked/redacted paths contain `*`?

### Are you interested in contributing a fix?

_No response_

--- a/src/main/java/edu/hm/hafner/analysis/parser/MsBuildParser.java
+++ b/src/main/java/edu/hm/hafner/analysis/parser/MsBuildParser.java
@@ -337,7 +337,12 @@ public class MsBuildParser extends LookaheadParser {
         if (fileName.contains("<") || fileName.contains(">") || fileName.contains("|")) {
             return true;
         }
-        return fileName.contains("\"") || fileName.contains("?") || fileName.contains("*");
+
+        if (fileName.contains("\"") || fileName.contains("?")) {
+            return true;
+        }
+
+        return fileName.contains("*") && !fileName.contains("/") && !fileName.contains("\\");
     }
 
     private boolean canResolveRelativeFileName(final String fileName, final String projectDir) {

--- a/src/test/java/edu/hm/hafner/analysis/parser/MsBuildParserTest.java
+++ b/src/test/java/edu/hm/hafner/analysis/parser/MsBuildParserTest.java
@@ -1180,6 +1180,35 @@ class MsBuildParserTest extends AbstractParserTest {
                 .hasMessage("ignoring '/OPT:REF' due to '/DEBUG' specification");
     }
 
+    /**
+     * MSBuildParser should keep warnings when Jenkins log masking replaces a directory segment with "****".
+     *
+     * @see <a href="https://github.com/jenkinsci/analysis-model/issues/1476">Issue 1476</a>
+     */
+    @Test
+    void issue1476() {
+        var withoutMask = parseStringContent(
+                "d:\\workspace\\foo\\Bar.cs(26,57): warning CS8625: Example warning text [d:\\workspace\\foo\\foo.csproj]");
+        assertThat(withoutMask).hasSize(1);
+
+        var withMask = parseStringContent(
+                "d:\\****\\workspace\\foo\\Bar.cs(26,57): warning CS8625: Example warning text [d:\\****\\workspace\\foo\\foo.csproj]");
+        assertThat(withMask).hasSize(1);
+
+        var warnings = parseStringContent(
+                "[2025-11-24T09:30:26.235Z]     34>d:\\****\\workspace\\foo\\Bar.cs(26,57): warning CS8625: Example warning text [d:\\****\\workspace\\foo\\foo.csproj]");
+
+        assertThat(warnings).hasSize(1);
+
+        assertThat(warnings.get(0))
+                .hasFileName("D:/****/workspace/foo/Bar.cs")
+                .hasCategory("CS8625")
+                .hasSeverity(Severity.WARNING_NORMAL)
+                .hasMessage("Example warning text")
+                .hasLineStart(26)
+                .hasColumnStart(57);
+    }
+
     @Override
     protected MsBuildParser createParser() {
         return new MsBuildParser();

